From 35a2c1524b21318228f0566d9a4ac53bf3138bdc Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Fri, 17 Feb 2017 09:47:11 +0000
Subject: [PATCH] BCM270X_DT: Add SMSC ethernet controller to DT

With an ethernet node in the DT, a suitable firmware can populate the
local-mac-address property, removing the need for a downstream patch
to the driver to read its MAC address from a module parameter.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/bcm2708-rpi-b-plus.dts | 1 +
 arch/arm/boot/dts/bcm2708-rpi-b.dts      | 1 +
 arch/arm/boot/dts/bcm2709-rpi-2-b.dts    | 1 +
 arch/arm/boot/dts/bcm2710-rpi-3-b.dts    | 1 +
 4 files changed, 4 insertions(+)

--- a/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts
+++ b/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts
@@ -1,6 +1,7 @@
 /dts-v1/;
 
 #include "bcm2708.dtsi"
+#include "bcm283x-rpi-smsc9514.dtsi"
 
 / {
 	model = "Raspberry Pi Model B+";
--- a/arch/arm/boot/dts/bcm2708-rpi-b.dts
+++ b/arch/arm/boot/dts/bcm2708-rpi-b.dts
@@ -1,6 +1,7 @@
 /dts-v1/;
 
 #include "bcm2708.dtsi"
+#include "bcm283x-rpi-smsc9512.dtsi"
 
 / {
 	model = "Raspberry Pi Model B";
--- a/arch/arm/boot/dts/bcm2709-rpi-2-b.dts
+++ b/arch/arm/boot/dts/bcm2709-rpi-2-b.dts
@@ -1,6 +1,7 @@
 /dts-v1/;
 
 #include "bcm2709.dtsi"
+#include "bcm283x-rpi-smsc9514.dtsi"
 
 / {
 	model = "Raspberry Pi 2 Model B";
--- a/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
@@ -5,6 +5,7 @@
 #endif
 
 #include "bcm2710.dtsi"
+#include "bcm283x-rpi-smsc9514.dtsi"
 
 / {
 	model = "Raspberry Pi 3 Model B";
